### PR TITLE
feat(cy.intercept): add `times` option to `RouteMatcher`

### LINKED ISSUE #4460: cy.route() unable to mock same url multiple times if requests happen quickly
### Current behavior:

If you make multiple requests to the same endpoint and want to mock each response differently based on the order that they occur, cypress appears to be unable to do it unless you wait X milliseconds between requests.  If there is no pause between requests, cypress will never catch the first `cy.wait(...)` and will mock every request to that endpoint with the same response.

### Desired behavior:

```js
cy.server()
cy.route(url, res1).as(get)
cy.wait(@get)
cy.route(url, res2).as(get)
```
Should result in the first GET request to url responding with res1 and the second GET request to url responding with res2, regardless of the amount of time between requests.

### Steps to reproduce: (app code and test code)

Run tests [here](https://github.com/bahalperin/cypress-route-bug-repro)

### LINKED ISSUE #8531: Add API for cy.intercept() to only reply to the first N requests
### Current behavior:
To change the response for a route over time, currently, the user must implement counting themselves:

```js
      let k = 0;
      cy.route2('/favorite-fruits', (req) => {
        k += 1
        switch (k) {
          case 1:
            return req.reply(['apples 🍎'])
          case 2:
            return req.reply(['grapes 🍇'])
          case 3:
            return req.reply(['kiwi 🥝'])
          default:
            return
        }
      })
```

### Desired behavior:
There is an API for the user to specify the number of times to use a route's handler. Something like this:

```js
cy.route2({ url: '/favorite-fruits', times: 1 }, ['apples 🍎'])
cy.route2({ url: '/favorite-fruits', times: 1 }, ['grapes 🍇'])
cy.route2({ url: '/favorite-fruits', times: 1 }, ['kiwi 🥝'])
```

### PR DESCRIPTION
**Pull request by flotwig**
- Closes #8531
- Closes #4460

### User facing changelog

- Added a `times` option to `cy.intercept`'s `RouteMatcher`. `times` specifies the number of times that a particular `cy.intercept` should be applied.

### How has the user experience changed?

New API:

```js
// this example will cause 1 request to `/temporary-error` to receive a network
// error. subsequent requests will not match this RouteMatcher.
cy.intercept('/temporary-error', { times: 1 }, { forceNetworkError: true })
```

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub? 
- [x] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? https://github.com/cypress-io/cypress-documentation/pull/3910
- [x] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/net-stubbing/lib/server/driver-events.ts [hunk 1]
```diff
@@ -84,8 +93,10 @@ export async function onNetEvent (opts: OnNetEventOpts): Promise<any> {
   debug('received driver event %o', { eventName, args })
 
   switch (eventName) {
-    case 'route:added':
-      return _onRouteAdded(state, getFixture, <NetEventFrames.AddRoute>frame)
+    case 'create:route':
+      return _createRoute(state, getFixture, <NetEventFrames.CreateRoute>frame)
+    case 'set:route:disabled':
```
Comment (flotwig) [inline packages/net-stubbing/lib/server/driver-events.ts:98]: ended up not going this route but i figure we may as well keep the method kicking around for prototyping cy intercept overrides

File: packages/driver/cypress/integration/commands/net_stubbing_spec.ts [hunk 2]
```diff
@@ -1170,6 +1180,31 @@ describe('network stubbing', { retries: 2 }, function () {
...
+              'second',
+              'fourth',
+              'fourth',
+              'fourth',
+            ], expectGet)
+
+            expect(third).to.be.calledTwice
+          })
+        })
+      })
```
Comment (flotwig) [inline packages/driver/cypress/integration/commands/net_stubbing_spec.ts:1207]: @brian-mann thoughts on this? I know you weren't originally the biggest fan of this API, but after some research, I think it's still the most direct way to accomplish the user ask

File: packages/driver/cypress/integration/commands/net_stubbing_spec.ts [hunk 3]
```diff
@@ -1170,6 +1180,31 @@ describe('network stubbing', { retries: 2 }, function () {
...
+            await Promise.mapSeries([
+              'first',
+              'second',
+              'second',
+              'fourth',
+              'fourth',
+              'fourth',
+            ], expectGet)
+
+            expect(third).to.be.calledTwice
```
Comment (bahmutov) [inline packages/driver/cypress/integration/commands/net_stubbing_spec.ts:1204]: can we assert that after the intercepts are exhausted they no longer apply?

Comment (flotwig) [reply packages/driver/cypress/integration/commands/net_stubbing_spec.ts:1204]: that is what this test is doing

Comment (flotwig) [reply packages/driver/cypress/integration/commands/net_stubbing_spec.ts:1204]: do you mean like via cy.wait?

Comment (bahmutov) [reply packages/driver/cypress/integration/commands/net_stubbing_spec.ts:1204]: I mean more like that the very last intercept `.intercept({ url: '/foo', times: 3 }, 'fourth')` is not longer used after 3 times, and the endpoint receives it.

**Review by bahmutov — APPROVED** [review]
trying the new option in https://github.com/cypress-io/cypress-example-recipes/pull/627. Much more convenient. For example, before in file cypress/integration/control-clock-spec.js without `cy.intercept times`

```js
// first request - respond with 3 fruits
// second request - respond with 4 fruits
let k = 0
const firstList = ['Apple', 'Banana', 'Cantaloupe']
const secondList = ['Orange', 'Cherry', 'Raspberry', 'Pineapple']

cy.intercept('/favorite-fruits', (req) => {
  k += 1
  if (k === 1) {
    req.reply(firstList)
  } else {
    req.reply(secondList)
  }
})
```

Using `times` options
```js
// first request - respond with 3 fruits
// after that always respond with 4 fruits
const firstList = ['Apple', 'Banana', 'Cantaloupe']
const secondList = ['Orange', 'Cherry', 'Raspberry', 'Pineapple']

cy.intercept({ pathname: '/favorite-fruits', times: 1 }, firstList)
cy.intercept({ pathname: '/favorite-fruits' },secondList)
```

file cypress/integration/stub-fetch-spec.js

before:
```js
it('can stub network calls for each page', () => {
  let k = 0

  cy.intercept('/favorite-fruits', (req) => {
    k += 1
    switch (k) {
      case 1:
        return req.reply(['apples 🍎'])
      case 2:
        return req.reply(['grapes 🍇'])
      default:
        return req.reply(['kiwi 🥝'])
    }
  })

  cy.visit('/')
  cy.contains('apples 🍎')
  ...
})
```

after:

```js
it('can stub network calls for each page', () => {
  // first time respond with an apple
  // second time respond with grapes
  // and after that always respond with kiwi
  cy.intercept({ pathname: '/favorite-fruits', times: 1 }, ['apples 🍎'])
  cy.intercept({ pathname: '/favorite-fruits', times: 1 }, ['grapes 🍇'])
  cy.intercept({ pathname: '/favorite-fruits' }, ['kiwi 🥝'])

  cy.visit('/')
  cy.contains('apples 🍎')
  ...
})
```

So the new syntax is much much simpler

File: packages/driver/cypress/integration/commands/net_stubbing_spec.ts (hunk 3 above)
Comment (flotwig) [reply packages/driver/cypress/integration/commands/net_stubbing_spec.ts:1584]: alright, i've added a check that a final request falls through, as well as asserting that `alias.all` matches the expected length

**Review by chrisbreiding — APPROVED** [review]
Very nice addition 💯

**Review by bahmutov — APPROVED** [review]
very nice

**Pull request merged**